__tests__/components/AnonymousChat/child/ChatItemMyTextV2.test.tsx

```diff
@@ -1,34 +1,173 @@
+    );
+
+    act(() => {
+      fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
+    });
+
+    expect(setReplyPreview).toHaveBeenCalled();
+  });
+
+  it('should display the username and time when isContinuous is false', () => {
+    const contextValue = {
+      chat: [{replyTarget: null}, jest.fn()]
+    };
+
+    const {getByText} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
+    );
+
+    expect(getByText('username')).toBeTruthy();
+    expect(getByText('time')).toBeTruthy();
+  });
+
+  it('should not display the username and time when isContinuous is true', () => {
+    const contextValue = {
+      chat: [{replyTarget: null}, jest.fn()]
+    };
+
+    const {queryByText} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={true}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
     );
 
-    expect(toJSON()).toMatchSnapshot();
+    expect(queryByText('username')).toBeNull();
+    expect(queryByText('time')).toBeNull();
   });
 
-  it('RENDER continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemMyTextV2
-        avatar="https://www.google.com"
-        username="username"
-        isContinuous={true}
-        message="message"
-        time="time"
-      />
+  it('should display the message', () => {
+    const contextValue = {
+      chat: [{replyTarget: null}, jest.fn()]
+    };
+
+    const {getByText} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
     );
 
-    expect(toJSON()).toMatchSnapshot();
+    expect(getByText('message')).toBeTruthy();
+  });
+
+  it('should match snapshot', () => {
+    const setReplyPreview = jest.fn();
+    const contextValue = {
+      chat: [{replyTarget: null}, setReplyPreview]
+    };
+
+    const tree = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Snapshot tests can be useful for detecting unexpected changes in your components, but they should be used judiciously. If your component has a lot of dynamic behavior or is likely to change frequently, snapshot tests can become brittle and add maintenance overhead. Make sure you're getting enough value out of this snapshot test to justify its potential cost.


<!-- This is an auto-generated comment by OSS CodeRabbit -->